sample/tv/src/main/java/com/datadog/android/tv/sample/net/OkHttpDownloader.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.tv.sample.net
+
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody
+import okhttp3.RequestBody.Companion.toRequestBody
+import org.schabi.newpipe.extractor.downloader.Downloader
+import org.schabi.newpipe.extractor.downloader.Response
+import org.schabi.newpipe.extractor.exceptions.ReCaptchaException
+import org.schabi.newpipe.extractor.downloader.Request as PipeRequest
+
+internal class OkHttpDownloader(
+    private val okHttpClient: OkHttpClient
+) : Downloader() {
+
+    override fun execute(request: PipeRequest): Response {
+        val httpMethod = request.httpMethod()
+        val url = request.url()
+        val headers = request.headers()
+        val dataToSend = request.dataToSend()
+
+        var requestBody: RequestBody? = null
+        if (dataToSend != null) {
+            requestBody = dataToSend.toRequestBody(null, 0, dataToSend.size)
+        }
+
+        val requestBuilder = Request.Builder()
+            .method(httpMethod, requestBody)
+            .url(url)
+            .addHeader("User-Agent", USER_AGENT)
+
+        headers.forEach { (name, valueList) ->
+            if (valueList.size > 1) {
+                requestBuilder.removeHeader(name)
+                for (headerValue in valueList) {
+                    requestBuilder.addHeader(name, headerValue)
+                }
+            } else if (valueList.size == 1) {
+                requestBuilder.header(name, valueList[0])
+            }
+        }
+
+        val response = okHttpClient.newCall(requestBuilder.build()).execute()
+
+        if (response.code == TOO_MANY_REQUESTS) {
+            response.close()
+            throw ReCaptchaException("reCaptcha Challenge requested", url)
+        }
+
+        val body = response.body
+        var responseBodyToReturn: String? = null
```

Comment:
minor: we can make it `val responseBodyToReturn = response.body?.let { it.string() }`

Reply:
👍